CONTRIBUTING.md

```diff
@@ -0,0 +1,374 @@
+
+- a commit should contain exactly one self-contained functional change
+- a functional change should be contained in exactly one commit
+- a commit should not create an inconsistent state (such as test errors, linting errors, partial fixes, feature with documentation, etc.)
+
+A complex feature can be broken down into multiple commits as long as each one maintains a consistent state and consists of a self-contained change.
+
+#### Commit message format
+
+Following these guidelines for committing messages is optional but strongly encouraged.
+
+Each commit message consists of a **header** and a **body**.
+The header has a special format that includes a **type**, a **scope** and a **subject**:
+
+```commit
+<type>(<scope>): <subject>
+<BLANK LINE>
+<body>
+```
+
+The **header** is mandatory and the **scope** of the header is optional.
+
+The **body** can contain a [closing reference to an issue](https://help.github.com/articles/closing-issues-via-commit-messages).
+
+#### Pull Request message format
+
+Following these guidelines for Pull Request messages is **required** and will be enforced by the CI. This is a strong prerequisite for our [release automation](#releasing-changes).
+
+Each pull request consists of a **title** and a **body**
+
+The title has a special format that includes a **type**, a **scope** and a **subject**:
+
+```commit
+<type>(<scope>): <subject>
+```
+
+#### Revert
+
+If the commit reverts a previous commit, it should begin with `revert: `, followed by the header of the reverted commit.
+The body should say: `This reverts commit <hash>.`, where the hash is the SHA of the commit being reverted.
+
+#### Type
+
+The type must be one of the following:
+
+| Type         | Description                                                                                            |
+| ------------ | ------------------------------------------------------------------------------------------------------ |
+| **fix**      | A bug fix                                                                                              |
+| **feat**     | A new feature                                                                                          |
+| **chore**    | A change that doesn't affect package codebase but rather the tooling around                            |
+| **docs**     | Documentation only changes                                                                             |
+| **style**    | Changes that do not affect the meaning of the code (white space, formatting, missing semicolons, etc.) |
+| **refactor** | A code change that neither fixes a bug nor adds a feature                                              |
+| **perf**     | A code change that improves performance                                                                |
+| **test**     | Adding missing tests or correcting existing tests                                                      |
+
```

Comment:
```suggestion
| **revert**     | Reverts a previous commit                                                      |
```
Should we include `revert` in this table as well since it's being added in the `pr-semantic` workflow in this PR?